### ISSUE TITLE
Read Only Connection / Transactions

### ISSUE THREAD

**Issue by Braden1996** [ospfranco/react-native-quick-sqlite#116]
Hey 👋🏻 

Great work on putting this library!
For my use case, one of the features I'm after are read-transactions. Is this possible?

Cheers,

**Comment by ospfranco** [comment]
Ah, that's what you meant, not really sure how to prevent users from writing inside of these transactions though... I guess you are better off writing your own sub-routine? you can just execute the sql commands directly and wrap everything inside normal transactions/promises, you can see from the internal code there is no magic to it.

**Comment by Braden1996** [comment]
> Ah, that's what you meant, not really sure how to prevent users from writing inside of these transactions though... I guess you are better off writing your own sub-routine? you can just execute the sql commands directly and wrap everything inside normal transactions/promises, you can see from the internal code there is no magic to it.
> 
> But feel free to submit a PR if you feel strongly about it.

I took a quick look at react-native-sqlite-storage and they seem to use a [regex to detect if the SQL](https://github.com/andpor/react-native-sqlite-storage/blob/b6aeac439b9d038bbb9d2e16812d512c38246775/lib/sqlite.core.js#L422) used within a transaction is read-only. I suppose that would work for most cases, but likely not all. 

Would it be possible to open a database connection in read-only mode? Perhaps in such case transactions can then safely run in parallel?

**Comment by ospfranco** [comment]
Not sure, I feel like that would increase the complexity too much and I barely have enough time as it is to maintain the package. In my mind the ideal solution would be checking the return values of the C code sqlite calls... but there is no escaping these transactions would need a separate typescript implementation. I guess there the regex is good enough for most cases. Feel free to sketch a PR, I will review it gladly and merge it if is ready.

**Comment by ospfranco** [comment]
If this is still relevant please open a new ticket in the new package: [op-sqlite](https://github.com/OP-Engineering/op-sqlite)
